ynr/apps/elections/templates/elections/ballot_view.html

```diff
@@ -31,92 +31,323 @@
+
+
+      {# Locking #}
+      {% if user_can_lock %}
+        <form method="post" action="{% url 'constituency-lock' ballot_id=ballot.ballot_paper_id %}">
+          {% csrf_token %}
+          {% if ballot.has_lock_suggestion and sopn %}
+            {% if current_user_suggested_lock %}
+               <p>
+                Locking disabled because you suggested locking this ballot.
+                Someone else will double check it soon.
+              </p>
             {% else %}
-            <p>Thanks, you've suggested we lock this list – we'll review it soon.</p>
+              <input type="submit" class="button small" value="Lock candidate list">
             {% endif %}
+
+          {% elif ballot.candidates_locked %}
+            <input type="submit" class="button small" value="Unlock candidate list">
+          {% endif %}
+
+          {% if ballot.candidates_locked %}
+            This list of candidates is currently <strong>locked</strong>.
+          {% else %}
+            This list of candidates is currently <strong>unlocked</strong>.
           {% endif %}
+        </form>
+      {% else %}
+        {% if candidates_locked %}
+          <p>
+            This list of candidates is now <strong>locked</strong>;
+            you can still update contact details of candidates, but
+            can't change the people standing in this constituency.
+          </p>
         {% endif %}
+      {% endif %}
 
+      {% if membership_edits_allowed %}
+        <p>
+          <a class="show-new-candidate-form button" href="{% url 'person-create' election=ballot.election.slug %}">
+            Add a new candidate
+          </a>
+        </p>
+        {% if sopn %}
+          <a href="{% url "bulk_add_from_sopn" ballot.election.slug ballot.post.id %}" class="button">
+            Add candidates from nomination paper in {{ballot.post.label}}
+          </a>
+        {% endif %}
+
+      {% elif not request.user.is_authenticated %}
+        <p>
+          <a href="{% url 'account_login' %}?next={{ request.path }}" class="show-new-candidate-form button">
+            Sign in to add a new candidate
+          </a>
+        </p>
+      {% endif %}
     </div>
-    <div class="columns large-3">
-        {% include "uk/data_timeline.html" %}
+
+    {% if membership_edits_allowed and add_candidate_form %}
+      <div class="candidates__new" {% if add_candidate_form.errors %}style="display: block"{% endif %}>
+        <h4>Add a new candidate</h4>
+
+        <form id="new-candidate-form" name="new-candidate-form" action="{% url 'person-create' election=ballot.election.slug %}" method="post">
+        {% with form=add_candidate_form identifiers_formset=identifiers_formset %}
+          {% include 'candidates/_person_form.html' %}
+        {% endwith %}
+        </form>
+      </div>
+    {% endif %}
+
+
+    {% if membership_edits_allowed %}
+
+
+    {% comment %}<h2>People not standing again</h2>
+
+    {{ people_not_standing }}
+{% endcomment %}
```

Comment:
is this code in the `{% comment %}{% endcomment %}` supposed to be in or out? If its out, can we just delete it?